sdk/formrecognizer/azure-ai-formrecognizer/src/main/java/com/azure/ai/formrecognizer/FormTrainingAsyncClient.java

```diff
@@ -0,0 +1,304 @@
+        try {
+            return withContext(context -> deleteModelWithResponse(modelId, context));
+        } catch (RuntimeException ex) {
+            return monoError(logger, ex);
+        }
+    }
+
+    Mono<Response<Void>> deleteModelWithResponse(String modelId, Context context) {
+        Objects.requireNonNull(modelId, "'modelId' cannot be null");
+
+        return service.deleteCustomModelWithResponseAsync(UUID.fromString(modelId), context)
+            .map(response -> new SimpleResponse<>(response, null));
+    }
+
+    // list models
+    // TODO (shawn) : #9976
+
+    private Function<PollingContext<OperationResult>, Mono<CustomFormModel>> fetchTrainingModelResultOperation() {
+        return (pollingContext) -> {
+            final UUID modelUid = UUID.fromString(pollingContext.getLatestResponse().getValue().getResultId());
+            return service.getCustomModelWithResponseAsync(modelUid, true)
+                .map(modelSimpleResponse -> toCustomFormModel(modelSimpleResponse.getValue()));
+        };
+    }
+
+    private Function<PollingContext<OperationResult>, Mono<PollResponse<OperationResult>>>
+        createTrainingPollOperation() {
+        return (pollingContext) -> {
+            PollResponse<OperationResult> operationResultPollResponse = pollingContext.getLatestResponse();
+            try {
+                UUID modelUid = UUID.fromString(operationResultPollResponse.getValue().getResultId());
+                return service.getCustomModelWithResponseAsync(modelUid, true)
+                    .flatMap(modelSimpleResponse ->
+                        processTrainingModelResponse(modelSimpleResponse, operationResultPollResponse));
+            } catch (HttpResponseException e) {
+                logger.logExceptionAsError(e);
+                return Mono.just(new PollResponse<>(LongRunningOperationStatus.FAILED, null));
+            }
+        };
+    }
+
+    private Function<PollingContext<OperationResult>, Mono<OperationResult>> getTrainingActivationOperation(
+        String sourcePath, boolean includeSubFolders, String filePrefix, boolean useLabelFile) {
+
+        TrainSourceFilter trainSourceFilter = new TrainSourceFilter().setIncludeSubFolders(includeSubFolders)
+            .setPrefix(filePrefix);
+        TrainRequest serviceTrainRequest = new TrainRequest().setSource(sourcePath).
+            setSourceFilter(trainSourceFilter).setUseLabelFile(useLabelFile);
+        return (pollingContext) -> {
+            try {
+                return service.trainCustomModelAsyncWithResponseAsync(serviceTrainRequest)
+                    .map(response ->
+                        new OperationResult(parseModelId(response.getDeserializedHeaders().getLocation())));
+            } catch (RuntimeException ex) {
+                return monoError(logger, ex);
+            }
+        };
+    }
+
+    private static Mono<PollResponse<OperationResult>> processTrainingModelResponse(
```

Comment:
Is this similar to the other poller helpers? Would there be any way to combine them into a generic helper?

Reply:
There are total of 2 such methods both switch case and also return different return types.